patch_subset/codepoint_map.cc

```diff
@@ -0,0 +1,96 @@
+#include "patch_subset/codepoint_map.h"
+
+#include "common/logging.h"
+#include "hb.h"
+#include "patch_subset/hb_set_unique_ptr.h"
+
+namespace patch_subset {
+
+void CodepointMap::Clear() {
+  encode_map.clear();
+  decode_map.clear();
+}
+
+void CodepointMap::AddMapping(hb_codepoint_t from, hb_codepoint_t to) {
+  encode_map[from] = to;
+  decode_map[to] = from;
+}
+
+StatusCode CodepointMap::FromProto(const CodepointRemappingProto& proto) const {
+  // TODO(garretrieger): implement me!
+  LOG(WARNING) << "Not implemeneted yet.";
+  return StatusCode::kUnimplemented;
+}
+
+StatusCode CodepointMap::ToProto(CodepointRemappingProto* proto) const {
+  CompressedListProto* codepoint_ordering = proto->mutable_codepoint_ordering();
+
+  int last_cp = 0;
+  for (unsigned int i = 0; i < encode_map.size(); i++) {
```

Comment:
considering you immediately assign i to an hb_codepoint_t can you just loop on it directly? - for (hb_codepoint_t cp_for_index = 0;...)

Reply:
No because cp_for_index is modified by the Decode call.